src/main/java/io/zeebe/monitor/zeebe/kafka/importers/KafkaIncidentImporter.java

```diff
@@ -0,0 +1,52 @@
+package io.zeebe.monitor.zeebe.kafka.importers;
+
+import io.camunda.zeebe.protocol.record.intent.IncidentIntent;
+import io.zeebe.monitor.entity.IncidentEntity;
+import io.zeebe.monitor.repository.IncidentRepository;
+import io.zeebe.monitor.zeebe.kafka.GenericRecord;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+
+@Component
+public class KafkaIncidentImporter {
+
+  @Autowired private IncidentRepository incidentRepository;
+
+  public void importIncident(final GenericRecord record) {
+
+    final IncidentIntent intent = IncidentIntent.valueOf(record.getIntent());
+    final long key = record.getKey();
+    final long timestamp = record.getTimestamp();
+
+    Map values = record.getValue();
+
+    final IncidentEntity entity =
+        incidentRepository
+            .findById(key)
+            .orElseGet(
+                () -> {
+                  final IncidentEntity newEntity = new IncidentEntity();
+                  newEntity.setKey(key);
+                  newEntity.setBpmnProcessId((String)values.get("bpmnProcessId"));
+                  newEntity.setProcessDefinitionKey(values.get("processDefinitionKey") != null ? ((Number)values.get("processDefinitionKey")).longValue() : 0);
```

Comment:
hazelcast incident importer (and other importers) doesn't seems to handle `null` for the fields. Why we need `null` handing here?

Reply:
we don't get data as a same format like hazelcast, We are not able to reuse same entity/dto. we have used generic class where we are getting attributes as a Map.